fedimint-api/src/config.rs

```diff
@@ -96,17 +104,33 @@ impl<G: DkgGroup> Dkg<G> {
             .map(|(g, h)| g + h)
             .collect();
 
-        dkg.commitments.insert(our_id, commit.clone());
-        let step = dkg.broadcast(DkgMessage::Commit(commit));
+        let hashed = dkg.hash(commit.clone());
+        dkg.commitments.insert(our_id, commit);
+        dkg.hashed_commits.insert(our_id, hashed);
+        let step = dkg.broadcast(DkgMessage::HashedCommit(hashed));
 
         (dkg, step)
     }
 
     /// Runs a single step of the DKG algorithm, processing a `msg` from `peer`
     pub fn step(&mut self, peer: PeerId, msg: DkgMessage<G>) -> DkgStep<G> {
         match msg {
+            DkgMessage::HashedCommit(hashed) => {
+                match self.hashed_commits.get(&peer) {
+                    Some(old) if *old != hashed => panic!("{} sent us two hashes!", peer),
+                    _ => self.hashed_commits.insert(peer, hashed),
+                };
```

Comment:
This could be `assert!(self.hashed_commits.insert(peer, hashed).is_none())` or a similar if clause for better panic messages.

Reply:
So actually I've noticed in practice that messages can be sent twice over the network, which cause that assertion to fail, this is why I must check for non-equality.

Reply:
Interesting! Shouldn't this be prevented? Maybe that's a bug in the network layer?

Reply:
I'm out of context, but since I casually looking at it: It's not possible to have an atomic transaction over both persistence and network message, so generally "at least once delivery" is usually used, which means duplicated messages must be handled and idempotently.